server/aesdsocket.c

```diff
@@ -0,0 +1,276 @@
+#include <netdb.h>
+#include <unistd.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <signal.h>
+
+
+int socket_fd,accept_return; //Socket fd and client connection
+int status; //Daemon return status
+
+//Structure for bind and accept
+struct addrinfo hints;
+struct addrinfo *servinfo;
+struct sockaddr_in client_addr;
+
+
+char *store_data=NULL; //To store data from receiver
+
+
+//Signal handler
+void signal_handler(int sig)
+{
+	if(sig==SIGINT)
+	{
+		syslog(LOG_INFO,"Caught SIGINT, leaving");
+	}
+	else if(sig==SIGTERM)
+	{
+		syslog(LOG_INFO,"Caught SIGTERM, leaving");
+	}
+	
+	unlink("/var/tmp/aesdsocketdata.txt"); //Deletes a file
+	
+	//Close socket and client connection
+	close(socket_fd);
+	close(accept_return);
+	
+	exit(0); //Exit success 
+}
+
+
+int main(int argc, char *argv[])
+{
+	int getaddr,bind_status,listen_status; //Variables to get address for bind, return value of bind and listen
+	ssize_t rec_status=1; //Return of recv function
+	socklen_t size=sizeof(struct sockaddr); 
+	char buff[100]; //temporary buffer to store receiver data
+	int fd_status;
+	ssize_t write_status; //Retur of write function
+	int send_status; //Return of send function
+	int i, total_bytes=0,packet_bytes_total=0; //Store total bytes received in a packet and total bytes received in a connection
+	bool write_flag=false; 
+	
+	openlog(NULL,LOG_PID, LOG_USER); //To setup logging with LOG_USER
+	
+	
+	//To start a daemon process
+	if((argc>1) && strcmp(argv[1],"-d")==0)
+	{
+		if(daemon(0,0)==-1)
+		{
+			syslog(LOG_ERR, "Couldn't enter daemon mode");
+			exit(1);
+		}
+	}
+	
+	//To check if a signal is received
+	if(signal(SIGINT,signal_handler)==SIG_ERR)
+	{
+		syslog(LOG_ERR,"SIGINT failed");
+		exit(2);
+	}
+	if(signal(SIGTERM,signal_handler)==SIG_ERR)
+	{
+		syslog(LOG_ERR,"SIGTERM failed");
+		exit(3);
+	}
+	
+	//create socket
+	socket_fd=socket(PF_INET, SOCK_STREAM, 0);
+	if(socket_fd==-1)
+	{
+		syslog(LOG_ERR, "Couldn't create a socket");
+		exit(4);
+	}
+	
+	//Get server address
+	hints.ai_flags=AI_PASSIVE; //Set this flag before passing to function
+	getaddr=getaddrinfo(NULL,"9000",&hints,&servinfo);
+	if(getaddr !=0)
+	{
+		syslog(LOG_ERR, "Couldn't get the server's address");
+		exit(5);
+	}
+	
+	//Bind
+	bind_status=bind(socket_fd,servinfo->ai_addr,sizeof(struct sockaddr));
+	if(bind_status==-1)
+	{
+		freeaddrinfo(servinfo); //Free the memory before exiting
+		syslog(LOG_ERR, "Binding failed");
+		exit(6);
+	}
+	
+	freeaddrinfo(servinfo); //Free this memory before next operations
+	
+	fd_status=open("/var/tmp/aesdsocketdata.txt", O_WRONLY | O_CREAT | O_TRUNC, S_IRWXU | S_IRWXG | S_IRWXO); //To create a file and give permissions to all
+	if (fd_status==-1)
+	{
+		syslog(LOG_ERR, "The file could not be created/found");
+		exit(7);
+	}
+	close(fd_status);
+	
+	//Infinite loop for server-client connections
+	while(1)
+	{
+	
+	//Listen
+	listen_status=listen(socket_fd,20); //Max queue of connections set to 20
```

Comment:
@shashankchandrasekaran I don't think you need listen in your while loop and at least one student noticed issues with this - see https://discord.com/channels/1062535844696707115/1079159844252176425/1079209265857835069